PureBasicDebugger/MemoryViewer.pb

```diff
@@ -3,23 +3,128 @@
 ;  Dual licensed under the GPL and Fantaisie Software licenses.
 ;  See LICENSE and LICENSE-FANTAISIE in the project root for license information.
 ;--------------------------------------------------------------------------------------------
+Global Dim MemoryViewer_Chars.s(31)
 
+#MEMORY_VIEW_TABLE_DATA_DEC = 0 
+#MEMORY_VIEW_TABLE_DATA_HEX = 1 
+#MEMORY_VIEW_TABLE_DATA_OCT = 2 
 
-Global Dim MemoryViewer_Chars.s(31)
+Global MemoryViewTableData.i = MemoryIsHex 
 
-; Wrappers to simplify the handling of the different variable types.
-;
+Procedure.s MemoryViewer_GetDataView(ViewTableData) 
+  Select ViewTableData 
+    Case #MEMORY_VIEW_TABLE_DATA_DEC 
+      ProcedureReturn "DEC" 
+    Case #MEMORY_VIEW_TABLE_DATA_HEX
+      ProcedureReturn "HEX"
+    Case #MEMORY_VIEW_TABLE_DATA_OCT 
+      ProcedureReturn "OCT"
+  EndSelect     
+EndProcedure 
+
+Procedure.s OCT(number.q)
+  Protected oct.s=Space(8*SizeOf(Character))
+  For a = 7 To 0 Step -1
+    PokeS(@oct+a*SizeOf(Character),Str(number & 7),SizeOf(Character),#PB_String_NoZero)
+    number >> 3
+  Next 
+  oct = Trim(oct,"0") 
+  If oct = "        " 
```

Comment:
The whole IDE use uppercase as first variable name, would be great to continue like that :)

Reply:
Ok I will try